packages/common/src/common.ts

```diff
@@ -89,6 +89,55 @@ export class Common {
     }
   }
 
+  /**
+   * Update the internal Common EIP params set. Existing values
+   * will get preserved unless there is a new value for a paramter
+   * provided with params.
+   *
+   * Example Format:
+   *
+   * ```ts
+   * {
+   *   1559: {
+   *     initialBaseFee: 1000000000,
+   *   }
+   * }
+   * ```
+   *
+   * @param params
+   */
+  updateParams(params: ParamsDict) {
+    for (const [eip, paramsConfig] of Object.entries(params)) {
+      if (!(eip in this._params)) {
+        this._params[eip] = { ...paramsConfig } // copy
+      } else {
+        this._params[eip] = { ...this._params[eip], ...params[eip] }
```

Comment:
Does / should this deep copy?

Reply:
Ah, yes, likely, thanks for catching. I actually thought this would (and I still think it does to some extend, but maybe not on the second level dicts? 🤔) But then I re-stumbled upon this in the test writing, where I tried the same and it actually didn’t.

I will keep this here to save CI but do in subsequent PR today (also about Common).

Reply:
That's fine! I have addressed two other places where the shallow copy seems to happen with comments.